pkg/controller/utils/clientwrapper.go

```diff
@@ -0,0 +1,84 @@
+package utils
+
+import (
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/openshift/aws-account-operator/pkg/localmetrics"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+// NewClientWithMetricsOrDie creates a new controller-runtime client with a wrapper which increments
+// metrics for requests by controller name, HTTP method, URL path, and HTTP status. The client will
+// re-use the manager's cache. This should be used in all controllers.
+func NewClientWithMetricsOrDie(log logr.Logger, mgr manager.Manager, controller string) client.Client {
+	// Copy the rest.Config as we want our round trippers to be controller-specific.
+	cfg := rest.CopyConfig(mgr.GetConfig())
+	AddControllerMetricsTransportWrapper(cfg, controller)
+
+	options := client.Options{
+		Scheme: mgr.GetScheme(),
+		Mapper: mgr.GetRESTMapper(),
+	}
+	c, err := client.New(cfg, options)
+	if err != nil {
+		log.Error(err, "Unable to initialize metrics-wrapped client")
```

Comment:
`log.Fatal` and drop the `os.Exit`

Reply:
`logr.Logger` doesn't have a `Fatal`, and I didn't want to break with the rest of the repo by using a different logger interface.